lib/corefoundation/extensions.rb

```diff
@@ -1,157 +1,171 @@
+      # @return [CF::Number]
+      def to_cf
+        CF::Number.from_f(self)
+      end
+    end
 
-# Ruby false class
-class FalseClass
-  # Returns a CF::Boolean object representing false
-  # @return [CF::Boolean]
-  def to_cf
-    CF::Boolean::FALSE
-  end
-end
+    # Patches for ruby Array
+    module Array
+      # Converts the Array to an immutable {CF::Array} by calling `to_cf` on each element it contains
+      # @return [CF::Number]
+      def to_cf
+        CF::Array.immutable(collect(&:to_cf))
+      end
+    end
 
-# Ruby String class
-class String
-  # Returns a {CF::String} or {CF::Data} representing the string.
-  # If {#binary?} returns true a {CF::Data} is returned, if not a {CF::String} is returned
-  # 
-  # If you want a {CF::Data} with the contents of a non binary string, use {#to_cf_data}
-  #
-  # @return [CF::String, CF::Data]
-  def to_cf
-    self.binary? ? self.to_cf_data : self.to_cf_string
-  end
+    # Patches for ruby TrueClass
+    module TrueClass
+      # Returns a CF::Boolean object representing true
+      # @return [CF::Boolean]
+      def to_cf
+        CF::Boolean::TRUE
+      end
+    end
 
-  # @!method binary?
-  #
-  # used to determine whether {#to_cf} should return a {CF::String} or a {CF::Data}. On ruby 1.9 and above this simply 
-  # checks whether the encoding is ascii-8bit or not.
-  # 
-  # On ruby 1.8.7
-  #
-  # - A string is binary if you call {#binary!} with the default argument of true
-  # - A string is not binary if you call {#binary!} with the argument false
-  # 
-  # If you have never called {#binary!} then a string is binary if Iconv does not think it is valid utf-8
-  # @return whether the string is handled as binary data or not
-  #
-  if '<3'.respond_to? :encoding
-    def binary?
-      encoding == Encoding::ASCII_8BIT
+    # Patches for ruby FalseClass
+    module FalseClass
+      # Returns a CF::Boolean object representing false
+      # @return [CF::Boolean]
+      def to_cf
+        CF::Boolean::FALSE
+      end
     end
-  else
-    def binary?
-      unless defined? @cf_is_binary
-        begin
-          ::Iconv.conv('UTF-8', 'UTF-8', self)
-          return false if self.frozen?
-          @cf_is_binary = false
-        rescue Iconv::IllegalSequence
-          return true if self.frozen?
-          @cf_is_binary = true
+
+    # Patches for ruby String
+    module String
+      # Returns a {CF::String} or {CF::Data} representing the string.
+      # If {#binary?} returns true a {CF::Data} is returned, if not a {CF::String} is returned
+      #
+      # If you want a {CF::Data} with the contents of a non binary string, use {#to_cf_data}
+      #
+      # @return [CF::String, CF::Data]
+      def to_cf
+        binary? ? to_cf_data : to_cf_string
+      end
+
+      # @!method binary?
+      #
+      # used to determine whether {#to_cf} should return a {CF::String} or a {CF::Data}. On ruby 1.9 and above this
```

Comment:
Now that we don't support these old releases can we clean this up?

Reply:
resolved in https://github.com/chef/corefoundation/pull/10/commits/8071e13d1f6e0cbd67b45b2b2f0a70db00a5e6ee